src/scripts/main.js

```diff
@@ -1,3 +1,578 @@
+
+    if (move.key === 'ArrowLeft') {
+      moveLeft();
+      refresh();
+    }
+
+    victory();
+
+    for (let i = 0; i < arrayCells.length; i++) {
+      for (let j = 0; j < arrayCells[i].length; j++) {
+        if (arrayCells[i][j] !== arrayCellsOld[i][j]) {
+          result = true;
+        }
+      }
+    }
+
+    if (result) {
+      addRandomValues();
+    }
+
+    scoreBoard.textContent = `${score}`;
+  }
+});
+
+function moveUp() {
+  for (let j = 3; j >= 0; j--) {
+    const merger = [true, true, true, true];
+
+    for (let i = 3; i >= 0; i--) { // переміщення по пустих значеннях
+      if (
+        i < 3
+        && arrayCells[i + 1][j] !== undefined
+        && arrayCells[i + 1][j] !== 0
+        && arrayCells[i][j] === 0
+      ) {
+        arrayCells[i][j] = arrayCells[i + 1][j];
+        arrayCells[i + 1][j] = 0;
+        continue;
+      }
+    }
+
+    for (let i = 3; i >= 0; i--) {
+      if (
+        i === 3
+        && arrayCells[i][j] !== 0
+        && arrayCells[i - 1][j] === 0
+        && arrayCells[i - 2][j] !== 0
+        && arrayCells[i - 3][j] === 0
+      ) {
+        arrayCells[i - 3][j] = arrayCells[i - 2][j];
+        arrayCells[i - 2][j] = arrayCells[i][j];
+        arrayCells[i - 1][j] = 0;
+        arrayCells[i][j] = 0;
+        continue;
+      }
+
+      if ( // Для переміщення трьох нерівних
+        i === 3
+        && arrayCells[i][j] !== 0
+        && arrayCells[i - 1][j] !== 0
+        && arrayCells[i - 2][j] !== 0
+        && arrayCells[i - 3][j] === 0
+      ) {
+        arrayCells[i - 3][j] = arrayCells[i - 2][j];
+        arrayCells[i - 2][j] = arrayCells[i - 1][j];
+        arrayCells[i - 1][j] = arrayCells[i][j];
+        arrayCells[i][j] = 0;
+        continue;
+      }
+
+      if ( // Для переміщення двох нерівних
+        i >= 2
+        && arrayCells[i - 2][j] === 0
+        && arrayCells[i - 1][j] !== 0
+        && arrayCells[i][j] !== 0
+        && arrayCells[i][j] !== arrayCells[i - 1][j]
+      ) {
+        arrayCells[i - 2][j] = arrayCells[i - 1][j];
+        arrayCells[i - 1][j] = arrayCells[i][j];
+        arrayCells[i][j] = 0;
+        continue;
+      }
+
+      //
+
+      if ( // Для додавання
+        i < 3
+        && arrayCells[i + 1][j] !== undefined
+        && arrayCells[i + 1][j] !== 0
+        && arrayCells[i][j] === arrayCells[i + 1][j]
+        && merger[i] === true
+        && merger[i + 1] === true
+      ) {
+        arrayCells[i][j] += arrayCells[i][j];
+        arrayCells[i + 1][j] = 0;
+        merger[i] = false;
+        score += arrayCells[i][j];
+        continue;
+      }
+    }
+
+    for (let i = 3; i >= 0; i--) { // переміщення по пустих значеннях
+      if (
+        i < 3
+        && arrayCells[i + 1][j] !== undefined
+        && arrayCells[i + 1][j] !== 0
+        && arrayCells[i][j] === 0
+      ) {
+        arrayCells[i][j] = arrayCells[i + 1][j];
+        arrayCells[i + 1][j] = 0;
+        continue;
+      }
+    }
+  }
+}
+
+function moveDown() {
```

Comment:
The function 'moveDown' has complex nested loops and conditionals. Consider breaking it down into smaller functions for better readability and maintainability.